DeusEx/Classes/DXRFixup.uc

```diff
@@ -0,0 +1,154 @@
+class DXRFixup expands DXRBase;
+
+function FirstEntry()
+{
+    local Actor a;
+    local ScriptedPawn p;
+
+    Super.FirstEntry();
+    
+    switch(dxr.localURL)
+    {
+        case "06_HONGKONG_TONGBASE":
+            foreach AllActors(class'Actor', a)
+            {
+                switch(string(a.Tag))
+                {
+                    case "AlexGone":
+                    case "TracerGone":
+                    case "JaimeGone":
+                        a.Destroy();
+                        break;
+                    case "Breakintocompound":
+                    case "LumpathPissed":
+                        Trigger(a).bTriggerOnceOnly = False;
+                        break;
+                    default:
+                        break;
+                }
+            }    
+            break;
+        case "06_HONGKONG_WANCHAI_MARKET":
+            foreach AllActors(class'Actor', a)
+            {
+                switch(string(a.Tag))
+                {
+                    case "DummyKeypad01":
+                        a.Destroy();
+                        break;
+                    case "BasementKeypad":
+                    case "GateKeypad":
+                        a.bHidden=False;
+                        break;    
+                    case "Breakintocompound":
+                    case "LumpathPissed":
+                        Trigger(a).bTriggerOnceOnly = False;
+                        break;
+                    default:
+                        break;
+                }
+            }
+            
+            break;
+        default:
+            break;
+    }
+}
+
+function AnyEntry()
+{
+    Super.AnyEntry();
+    
+    doFixup();
+    
+}
+
+function ReEntry()
+{
+    Super.ReEntry();
+    doFixup();
+}
+
+function doFixup()
+{
+    local Actor a;
+    local ScriptedPawn p;
+    local name flagName;
+    local bool boolFlag;
+    
+    switch(dxr.localURL)
+    {
+        case "06_HONGKONG_TONGBASE":
+            flagName = dxr.Player.rootWindow.StringToName("QuickLetPlayerIn");
+            boolFlag = dxr.Player.flagBase.GetBool(flagName);
+            foreach AllActors(class'Actor', a)
+            {
+                switch(string(a.Tag))
+                {
+                    case "TriadLumPath":
+                        ScriptedPawn(a).ChangeAlly(dxr.Player.Alliance,1,False);
+                        break;
+                        
+                    case "TracerTong":
+                    case "AlexJacobson":
+                    case "JaimeReyes":                        
```

Comment:
what if you tell JaimeReyes to stay in UNATCO?

Reply:
That's actually the only scenario I tested, since I guess that's the default state of the flags.  As I have it right now, he doesn't show up if you directly load that map.  I'll try to track down the correct flag to have him show up and see if I need to add any extra logic for that.